### ISSUE TITLE
Index of EMBEDDEDLIST does not update properly

### ISSUE THREAD

**Issue by optimuspaul** [orientechnologies/orientdb#4325]
Create a basic class with some indexes, one of which is of an EMBEDDEDLIST.

```
CREATE CLASS i_have_a_list EXTENDS V
CREATE PROPERTY i_have_a_list.id STRING
CREATE INDEX i_have_a_list.id ON i_have_a_list (id) UNIQUE
CREATE PROPERTY i_have_a_list.types EMBEDDEDLIST STRING
CREATE INDEX i_have_a_list.types ON i_have_a_list (types) NOTUNIQUE
```

Create a vertex with some values.
```
CREATE VERTEX i_have_a_list CONTENT {"id": "the_id", "types": ["one", "two"]}
```
Verify it is there
```
SELECT * FROM i_have_a_list WHERE types = 'one'
```
Change that EMBEDDEDLIST a bit and update the vertex
```
UPDATE i_have_a_list CONTENT {"id": "the_id", "types": ["seven", "two"]} WHERE id = 'the_id'
```
Verify it changed
```
SELECT * FROM i_have_a_list WHERE types = 'one'
=> {'id': 'the_id', 'types': ['two', 'seven']}

SELECT * FROM i_have_a_list WHERE types = 'seven'
=> {'id': 'the_id', 'types': ['two', 'seven']}
```
Didn't get what I expected.  If I rebuild all indexes it gets it right, but that is not a good workaround. I can't rebuild indexes every time a vertex changes.

**Comment by lvca** [comment]
What OrientDB version?

**Comment by optimuspaul** [comment]
2.0.10

**Comment by luigidellaquila** [comment]
fixed in 2.0.x branch

**Comment by laa** [comment]
@luigidellaquila may be we will port in in 2.1 ?

**Comment by luigidellaquila** [comment]
@laa in 2.1 it seems to work fine

**Comment by optimuspaul** [comment]
Are their tests in 2.1 that prove it works?

**Comment by lvca** [comment]
Look at the commit: https://github.com/orientechnologies/orientdb/commit/7c735d17f3aa873e295e8a0cfe8a95e7a15c205b

**Comment by optimuspaul** [comment]
Will that be merged into the 2.1 release?

**Comment by luigidellaquila** [comment]
Hi @optimuspaul 

yes, I'll port it to 2.1

**Comment by lvca** [comment]
@luigidellaquila did you already port it to 2.1?

**Comment by luigidellaquila** [comment]
Just ported the test case, it passes correctly.
There was no need to also port the fix, because in 2.1 the management of dirty fields is different and the problem does not occur.
